repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -3585,6 +3495,25 @@ public List<String> propagateClassification(String entityGuid, String classifica
             AtlasVertex currentTaskVertex = (AtlasVertex) graph.query().has(TASK_GUID, currentTask.getGuid()).vertices().iterator().next();
             currentTaskVertex.setProperty(TASK_ASSET_COUNT_TO_PROPAGATE, impactedVertices.size());
 
+
+            //iterate over each impacted vertex to create a message and send it to kafka
+            for (AtlasVertex vertex : impactedVertices) {
+                // Create a Map to store vertex properties
+                Map<String, Object> vertexMap = new HashMap<>();
+                ObjectMapper objectMapper = new ObjectMapper(); // Jackson ObjectMapper
+
+                vertexMap.put("parentTaskVertexId", vertex.getProperty("parentTaskVertexId", String.class));
+                vertexMap.put("action", "ADD_PROPAGATION");
+                vertexMap.put("assetVertexId", vertex.getIdForDisplay());
+                vertexMap.put("tagVertexId", classificationVertexId);
+                vertexMap.put("parentTaskGuid", currentTask.getGuid());
+                vertexMap.put("tagTypeName", currentTask.getClassificationTypeName());
+
+                String vertexJson = objectMapper.writeValueAsString(vertexMap);
+
+                //send vertexJson to kafka topic 'TAG_PROP_EVENTS'
+                kfknotif.sendInternal(NotificationInterface.NotificationType.EMIT_PLANNED_RELATIONSHIPS, Collections.singletonList(vertexJson));
```

Comment:
We also need to specify the message key based on the asset GUID
Where are we sending message key for notification?

Regarding Kafka topic configurations, 
How many partition have been created?

Reply:
This has been addressed and updated in the PR. I'm resolving this comment, @nikhilbonte21.